src/libcore/option.rs

```diff
@@ -411,7 +411,9 @@ impl<T> Option<T> {
     /// ```
     /// let num_as_str: Option<String> = Some("10".to_string());
     /// // `Option::map` takes self *by value*, consuming `num_as_str`
```

Comment:
I was so tempted to make things more simple by just using `&str`, but decided to leave things alone, because maybe that point is important enough.